### ISSUE TITLE
WSL2 distro not starting since 21364 build

### ISSUE THREAD

**Issue by jpathy** [microsoft/WSL#6839]
### Windows Build Number

Microsoft Windows [Version 10.0.21364.1]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16

### Distro Version

Ubuntu 20.04

### Repro Steps

Start ubuntu. After the new insider release it doesn't work with following error.
````
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80070057
Error: 0x80070057 The parameter is incorrect.

Press any key to continue...
````

### Expected Behavior

Doesn't work.

### Actual Behavior

Should work.

### Diagnostic Logs

[wsl.zip](https://github.com/microsoft/WSL/files/6355982/wsl.zip)

**Comment by github348** [comment]
I have enabled wslg and wsl stopped working I have reported this issue to wslg repository 
you can fix wsl by adding this to your .wslconfig file 
[wsl2]
guiapplications=false

**Comment by anatol06** [comment]
I have the same issue. After update to Windows 10 Insider Preview Build 21364 WSL does start any more.
In terminal appears "The parameter is incorrect."

**Comment by waundme** [comment]
Same issue here. After update to Windows 10 Insider Preview Build 21364 WSL does start any more. 

[wsl2]
guiapplications=false

fixes the problem.

**Comment by slickag** [comment]
> I have enabled wslg and wsl stopped working I have reported this issue to wslg repository
> you can fix wsl by adding this to your .wslconfig file
> [wsl2]
> guiapplications=false

I was having the same issue and this fixed it for me.

Thanks!

**Comment by anatol06** [comment]
> Same issue here. After update to Windows 10 Insider Preview Build 21364 WSL does start any more.
> 
> [wsl2]
> guiapplications=false
> 
> fixes the problem.

Thanks!
It works now.

**Comment by nsaritzky** [comment]
@github348's fix worked for me, too.

For what it's worth, the the fix [https://github.com/microsoft/WSL/issues/5092#issuecomment-743937383](here) didn't work for me, nor did any of the other suggestions on that issue. I couldn't install any more distros; they all failed with error 0x80070057.

**Comment by renanlido** [comment]
[WSL2 - Windows 19041.264 - WslRegisterDistribution fails with error 0x80070057 - Incorrect parameter #5293](https://github.com/microsoft/WSL/issues/5293) 
apparently solves this problem as well.

**Comment by kudoshiji** [comment]
> I have enabled wslg and wsl stopped working I have reported this issue to wslg repository
> you can fix wsl by adding this to your .wslconfig file
> [wsl2]
> guiapplications=false

Worked for me.

**Comment by nwaltham** [comment]
This fix also works for me, although the reason I was trying to install it, was to try out the gui support., so I guess this means I can't

> > I have enabled wslg and wsl stopped working I have reported this issue to wslg repository
> > you can fix wsl by adding this to your .wslconfig file
> > [wsl2]
> > guiapplications=false
> 
> Worked for me.

This fix also works for me, although the reason I was trying to install it, was to try out the gui support., so I guess this means I can't

**Comment by adrastogi** [comment]
More details about the underlying cause can be found [here](https://github.com/microsoft/wslg/issues/33#issuecomment-825249237) and [here](https://github.com/microsoft/wslg/issues/33#issuecomment-825683356). A fix is being worked on.

**Comment by norberto-enomoto** [comment]
> I have enabled wslg and wsl stopped working I have reported this issue to wslg repository
> you can fix wsl by adding this to your .wslconfig file
> [wsl2]
> guiapplications=false

Thank you. I have almost lost entired night trying to solve this issue. Now it's up and running

**Comment by guradia** [comment]
On a german Windows Version it's simply `Falscher Paramter`, on might think it's a different problem, but this fix does the trick.

And it does not hinder any previous settings that let me use gui apps through X410.

Though i just notice, that gitkraken now can display it's icon (rather than the one of X410, that was used before the newest updates). May be unrelated, since i updated the WSL-Kernel as part of finding a solution.

**Comment by onomatopellan** [comment]
See reason of why this could happen in https://github.com/microsoft/wslg/issues/33

**Comment by benhillis** [comment]
We've identified an issue with some CPUs and a fix is inbound.

**Comment by P4R4DiSi4C** [comment]
@benhillis could this be also related to your fix ? [https://github.com/MicrosoftDocs/WSL/issues/798](https://github.com/MicrosoftDocs/WSL/issues/798)

**Comment by onomatopellan** [comment]
@P4R4DiSi4C To know if it's related just use [this wslconfig option](https://github.com/microsoft/WSL/issues/6839#issuecomment-825008988) to disable GUI support.
Also remember to run `wsl.exe --shutdown` to make the changes effective.

**Comment by P4R4DiSi4C** [comment]
> @P4R4DiSi4C To know if it's related just use [this wslconfig option](https://github.com/microsoft/WSL/issues/6839#issuecomment-825008988) to disable GUI support.
> Also remember to run `wsl.exe --shutdown` to make the changes effective.

Okay, then it's not related thanks :)

**Comment by kinglionsz** [comment]
> On a german Windows Version it's simply `Falscher Paramter`, on might think it's a different problem, but this fix does the trick.
> 
> And it does not hinder any previous settings that let me use gui apps through X410.
> 
> Though i just notice, that gitkraken now can display it's icon (rather than the one of X410, that was used before the newest updates). May be unrelated, since i updated the WSL-Kernel as part of finding a solution.

Yes, it's a good idea👍.  wsl2  ver.5.10.16.3 works  @win10 ver. 21359, but not work when update 21364!
thunar and gedit icon now can display on taskbar.👍

tks.

![wsl2-5 10 16 3-@win10-21359-gui-ok](https://user-images.githubusercontent.com/51827261/116348765-403c1a80-a821-11eb-8d2f-4dd4b315e4a1.png)

**Comment by MaximoTrinidad** [comment]
This need more investigation...  
I'm suspecting that this whole issue was due to a missing ".wsl2config" file. I'm only wondering this could have work by just creating an empty  file could. 
Also, we got to be on the lookout with upcoming Windows Insider updates breaking WSL installations.
I have an idea which I'm going to post soon to test this scenario.

For now... creating the ".wsl2config" file with the content:
```
[wsl2]
guiapplications=false
```
It Works!!

**Comment by MaximoTrinidad** [comment]
To repo this situation: (*** Please do this on a VM ***)

Duplicate the situation, now that I got my WSL to work , I'm went back to delete the config file: ".wslconfig" in the User Profile folder, run cmd "wsl --shutdown".  Reboot not necessary.

Start WSL and now I'm back to square one! So, I can't open WSL.

![WSLbad_2021-04-29_12-36-33](https://user-images.githubusercontent.com/19892435/116586611-a8edca80-a8e7-11eb-9ef1-b6e22971c1de.jpg)

Then try to recover with the following WSL Settings Options:
I did the "Repair Option", didn't work.
I did the "Reset Option, that didn't work neither.

I did uninstall all distro, remove the WSL Update, uncheck WSL Feature, then Reboot machine

Check back the WSL Feature, Reboot machine, install back the WSL Update then manually create the WSL config file "wslconfig" in your Windows User Profile folder" c:\Users\username" with the following entries:
```
[wsl2]
guiapplications=false
```

![wslconfig_2021-04-29_12-41-35](https://user-images.githubusercontent.com/19892435/116587276-54971a80-a8e8-11eb-98d3-b958202aef70.jpg)

Proceed to Download and Install UBUNTU-20.04
Open WSL and it's working again!!

Bottom-line: WSL needs to have the configuration file ".wslconfig" , or it won't start. So, you didn't need to uninstall WSL to get it to work.

![WSLOK_2021-04-29_12-33-13](https://user-images.githubusercontent.com/19892435/116586259-409ee900-a8e7-11eb-8e73-60ef314c4e37.jpg)

**Comment by benhillis** [comment]
This issue is well-understood.  Unless something unexpected happens this should be fixed with next week's Insider Dev build.

**Comment by andalou** [comment]
The fix mentioned here didn't work for me. After putting
guiapplications=false into %userprofile%\.wslconfig I get another error:

{Application Error}
The exception s (0xPress any key to continue...

**Comment by MaximoTrinidad** [comment]
@andalou 

Did you do open a PowerShell console (as Admin), and ran the  command: "wsl --shutdown"?
Then you should try to start wsl again.

**Comment by andalou** [comment]
> @andalou
> 
> Did you do open a PowerShell console (as Admin), and ran the command: "wsl --shutdown"?
> Then you should try to start wsl again.

Yes I did it, I get the same message

**Comment by andalou** [comment]
> @P4R4DiSi4C To know if it's related just use [this wslconfig option](https://github.com/microsoft/WSL/issues/6839#issuecomment-825008988) to disable GUI support.
> Also remember to run `wsl.exe --shutdown` to make the changes effective.

I already did it. I put
[wsl2]
guiapplications=false

into %userprofile%\\.wslconfig and I did, as administrator, wsl.exe --shutdown but I get another message, namely:
{Application Error}
The exception s (0xPress any key to continue...

That is 21364.1 build

**Comment by MaximoTrinidad** [comment]
@andalou 

Please, can you provide a print-screen of the message you're getting?
thx

**Comment by andalou** [comment]
> @andalou
> 
> Please, can you provide a print-screen of the message you're getting?
> thx

![ubuntu1](https://user-images.githubusercontent.com/352173/116756825-681eb000-a9d2-11eb-8a91-741f8b8cc3f1.jpg)

**Comment by MaximoTrinidad** [comment]
@andalou 

That message, I haven't seen it.  I ended up with the pain of resetting and reinstalling everything (all components/features) to finally get it to work.

Today I updated my Windows 10 Insider to Build 21370.

This time WSL works! But now fighting with Docker-Desktop(3.3.1), which installs but doesn't work in WSL2.
I hope someone can offer a resolution.

**Comment by onomatopellan** [comment]
@MaximoTrinidad I'm running it right now without problem. Try adding `guiapplications=false` to your .wslconfig file and `wsl.exe --shutdown` before launching Docker Desktop.

**Comment by anatol06** [comment]
> @andalou
> 
> That message, I haven't seen it. I ended up with the pain of resetting and reinstalling everything (all components/features) to finally get it to work.
> 
> Today I updated my Windows 10 Insider to Build 21370.
> 
> This time WSL works! But now fighting with Docker-Desktop(3.3.1), which installs but doesn't work in WSL2.
> I hope someone can offer a resolution.

My advice regarding Docker is to install the docker on WSL directly.

**Comment by mayurigohil** [comment]
> I have enabled wslg and wsl stopped working I have reported this issue to wslg repository
> you can fix wsl by adding this to your .wslconfig file
> [wsl2]
> guiapplications=false

This works for me.

After so many Trials finally Docker is running.

Thanks so much

**Comment by gmag11** [comment]
Fixed adding .wslconfig file to %UserProfile%

```
PS C:\WINDOWS\system32> wsl --set-default-version 2
For information on key differences with WSL 2 please visit https://aka.ms/wsl2
The operation completed successfully.
PS C:\WINDOWS\system32> wsl --status;cmd /c ver
Default Distribution: Ubuntu-20.04
Default Version: 2

Windows Subsystem for Linux was last updated on 02/05/2021
WSL automatic updates are on.

Kernel version: 5.10.16

Microsoft Windows [Version 10.0.21370.1]
```

**Comment by kinglionsz** [comment]
> Fixed adding .wslconfig file to %UserProfile%. Thank you
> 
> ```
> PS C:\WINDOWS\system32> wsl --set-default-version 2
> For information on key differences with WSL 2 please visit https://aka.ms/wsl2
> The operation completed successfully.
> PS C:\WINDOWS\system32> wsl --status;cmd /c ver
> Default Distribution: Ubuntu-20.04
> Default Version: 2
> 
> Windows Subsystem for Linux was last updated on 02/05/2021
> WSL automatic updates are on.
> 
> Kernel version: 5.10.16
> 
> Microsoft Windows [Version 10.0.21370.1]
> ```

TKS.  I will have a try.

**Comment by aarav2you** [comment]
> I have enabled wslg and wsl stopped working I have reported this issue to wslg repository
> you can fix wsl by adding this to your .wslconfig file
> [wsl2]
> guiapplications=false

This did not do anything for me.. Restarted machine & WSL and reinstalled WSL with the `.wslconfig` file but it didn't work!

**Comment by bpulliam** [comment]
The fix for this issue (not a workaround) is in build 21376.

**Comment by kinglionsz** [comment]
> Fixed adding .wslconfig file to %UserProfile%. Thank you
> 
> ```
> PS C:\WINDOWS\system32> wsl --set-default-version 2
> For information on key differences with WSL 2 please visit https://aka.ms/wsl2
> The operation completed successfully.
> PS C:\WINDOWS\system32> wsl --status;cmd /c ver
> Default Distribution: Ubuntu-20.04
> Default Version: 2
> 
> Windows Subsystem for Linux was last updated on 02/05/2021
> WSL automatic updates are on.
> 
> Kernel version: 5.10.16
> 
> Microsoft Windows [Version 10.0.21370.1]
> ```

```
The parameter is incorrect.

[process exited with code 4294967295]
```

**Comment by bpulliam** [comment]
Build 21376 releases today.  We encourage you to turn GUI applications back on, as that should now be fixed (you can remove guiapplications=false from .wslconfig).

**Comment by dogansan** [comment]
> Build 21376 releases today. We encourage you to turn GUI applications back on, as that should now be fixed (you can remove guiapplications=false from .wslconfig).

Confirmed, GUI is now working

**Comment by vuon9** [comment]
> > Build 21376 releases today. We encourage you to turn GUI applications back on, as that should now be fixed (you can remove guiapplications=false from .wslconfig).
> 
> Confirmed, GUI is now working

Can u give me a link or the way to enable GUI?

**Comment by dogansan** [comment]
> > > Build 21376 releases today. We encourage you to turn GUI applications back on, as that should now be fixed (you can remove guiapplications=false from .wslconfig).
> > 
> > 
> > Confirmed, GUI is now working
> 
> Can u give me a link or the way to enable GUI?

https://github.com/microsoft/wslg

**Comment by kinglionsz** [comment]
Confirmed, GUI is now working Update 21376. Tks.

![wsl-ok-21376-Screenshot 2021-05-07 123233](https://user-images.githubusercontent.com/51827261/117435494-82233a00-af60-11eb-97ad-b8dbf359a9bf.png)

**Comment by MaximoTrinidad** [comment]
In my case, after 3hrs installing Windows Insider build 21376, on my old Lenovo T430s, and removing the line:
"guiapplications=false
" on my .wslcinfig file ., I got an error two times: 
"Windows SUBSystem for Linus was terminated.

I had to use the command in the PowerShell console:

```
wsl --shutdown
```

`Then, I was able to enable open  WSL2with graphics enabled.

What a ride!

I'm glad this is finally resolved!

![WSL2GUI_OK_2021-05-07_8-44-43](https://user-images.githubusercontent.com/19892435/117452406-b9b6c580-af11-11eb-8f60-d7a0d773ac90.jpg)
